### PR TITLE
fix: throw error when any additional path doesn't exist

### PR DESCRIPTION
**Pull request by jgjadaoag**
#### :rocket: Why this change?

It seems that dredd doesn't show an error when there is a path that doesn't exist which is not the expected behavior.

#### :memo: Related issues and Pull Requests

Fixes https://github.com/apiaryio/dredd/issues/378

#### :white_check_mark: What didn't I forget?

- [x] To write docs
- [x] To write tests
- [x] To put [Conventional Changelog](https://dredd.readthedocs.io/en/latest/contributing/#sem-rel) prefixes in front of all my commits and run `npm run lint`

### CODE REVIEW

**Comment by michalholasek** [comment]
Hello @jgjadaoag, thank you for the PR! Looks good, I have just one nitpick - could you please change the error message to something like `No API Description document(s) were found on path:`? I think that it sounds a little bit better than the one you used. What do you think?

**Comment by honzajavorek** [comment]
This may be more complicated problem then it seems. Glob means we have no idea how many files were supposed to be loaded. We can probably assume the user wanted Dredd to load 1 or more files. Also, loading hook files has probably the same issues.

I suggest we make sure (= write tests) that for each given path, if the path evaluates to zero files, Dredd ends with an error. And I suggest we do this for both the first argument (API description file), the `--path` option, and the `--hookfiles` option.

**Comment by jgjadaoag** [comment]
@michalholasek yup, that looks better, I just copied the error message from a similar error.

@honzajavorek got it, it seems that there are already unit tests and integration tests that test the opposite (that dredd shouldn't fail if a path evaluates to zero files) which I am changing to accommodate the new behavior. I'll try and push a commit today or tomorrow for the hookfiles.

**Comment by michalholasek** [comment]
@jgjadaoag I am sorry for the delay, we were buried in other work.

Unfortunately, last change does not work as expected (and tests you added fail aswell) because it is enclosed in a callback function of `hookfiles.reduce` method (`return callback(new Error(...))` is a return value of `hookfiles.reduce`, not `addHooks`).

I think something like

```diff
diff --git a/src/add-hooks.coffee b/src/add-hooks.coffee
index 11e2e7d..d9251c1 100644
--- a/src/add-hooks.coffee
+++ b/src/add-hooks.coffee
@@ -43,7 +43,7 @@ addHooks = (runner, transactions, callback) ->
       fixLegacyTransactionNames runner.hooks
 
     catch error
-      logger.warn("""\
+      logger.error("""\
         Skipping hook loading. Error reading hook file '#{filePath}'. \
         This probably means one or more of your hook files are invalid.
         Message: #{error.message}
@@ -105,9 +105,6 @@ addHooks = (runner, transactions, callback) ->
       # glob.sync does not resolve paths, only glob patterns
       unresolvedPaths = if glob.hasMagic(unresolvedPath) then glob.sync(unresolvedPath) else [unresolvedPath]
 
-      if unresolvedPaths.length == 0
-        return callback(new Error("Hook file(s) not found on path: #{unresolvedPaths}"))
-
       # Gradually append sorted and resolved paths
       result.concat unresolvedPaths
         # Create a filename / filepath map for easier sorting
@@ -129,6 +126,11 @@ addHooks = (runner, transactions, callback) ->
     , [] # Start with empty result
     )
 
+    if files.length is 0
+      message = "No hookfiles found on path(s) #{hookfiles.join(', ')}"
+      logger.error(message)
+      return callback(new Error(message))
+
     logger.info('Found Hookfiles:', files)
 
     # Clone the configuration object to hooks.configuration to make it
```

could be a good solution. Would you agree?

**Comment by jgjadaoag** [comment]
@michalholasek Sorry for the late reply, I have been busy as well the past few weeks. I think that solution won't work for cases when there is a mix of hookfiles which have a valid and invalid path. 

I pushed an improvement to my fix and I also corrected my tests.

File: src/add-hooks.coffee [hunk 1]
```diff
@@ -126,6 +132,9 @@ addHooks = (runner, transactions, callback) ->
     , [] # Start with empty result
     )
 
+    if files instanceof Error
+      return callback(files)
+
```
[the remark this answers is not in the document]
Comment (jgjadaoag) [reply src/add-hooks.coffee]: Ok, got it.

File: test/integration/cli/api-description-cli-test.coffee [hunk 2]
```diff
@@ -268,7 +268,7 @@ describe 'CLI - API Description Document', ->
           runtimeInfo = info
           done(err)
 
-      it 'should request /machines', ->
-        assert.deepEqual runtimeInfo.server.requestCounts, {'/machines': 1}
-      it 'should exit with status 0', ->
-        assert.equal runtimeInfo.dredd.exitStatus, 0
+      it 'should exit with status 1', ->
```
[the remark this answers is not in the document]
Comment (jgjadaoag) [reply test/integration/cli/api-description-cli-test.coffee]: Ok, I also changed my implementation to make sure that every hook file is available and it seems to have affected multiple tests. I'll try and fix them later or on the weekends.

**Pull request closed without merging**